dataservice/api/participant/models.py

```diff
@@ -14,3 +23,7 @@ class Participant(db.Model, Base):
     """
     __tablename__ = "participant"
     external_id = db.Column(db.String(32))
+    diagnoses = db.relationship('Diagnosis', secondary=diagnoses_assoc_table,
+                                lazy='joined',
+                                backref=db.backref('participants',
+                                                   lazy=True))
```

Comment:
**Using lazy=True for Diagnosis.participants**
In terms of the reverse relationship (how to load participants for a diagnosis or diagnoses), it is tough to tell what to choose here because this will be dependent on the query use cases. For example, when a user queries for a diagnosis will they usually just be interested in the diagnosis shallow attributes (i.e. primary diagnosis) or will they usually also be interested in getting all the participants or a subset of participants (perhaps based on some condition) for that diagnosis. In another example, when a user queries for a collection of diagnoses will they be interested in just the diagnoses shallow attributes for the collection of diagnoses or will they usually want to query for a collection of diagnoses related to a set of participants that meet some condition(s). In each of these examples the lazy loading parameter can make a significant impact on performance. Since this is unclear right now, I am leaving the lazy loading parameter to the default (lazy=True) which simply loads participants when they are needed rather than up front (lazy=joined or lazy=subquery). We can always change this later once we start getting into implementing richer query functionality.

Reply:
Again, the resource fetching functionality can be set on a per-query basis, so we can set them depending on the needs of whatever is querying.

Reply:
👍